### ISSUE TITLE
Directory-contextual results

### ISSUE THREAD

**Issue by NellyWhadsDev** [zdharma/history-search-multi-word#15]
TLDR: Integrated, directory-based results, with a toggle (CTRL+G perhaps? 😉).

Context: I know this is possible via [this plugin](https://github.com/robbyrussell/oh-my-zsh/tree/master/plugins/per-directory-history), however, using that plugin breaks [this one](https://github.com/djui/alias-tips). I haven't been able to trace down the source of the is one. Alias suggestions are never shown when both are used in conjunction. I've tried [this plugin as well](https://github.com/MichaelAquilina/zsh-you-should-use) with the same result.

Request: How difficult would it be to build this into history-search-multi-word? If you do not have time for the feature, I can try to implement it as a branch here; I've just never written a zsh plugin before.

**Comment by psprint** [comment]
I actually do have some time issues, so any help would be appreciated

pt., 19 lip 2019, 19:25 użytkownik NellyWhads <notifications@github.com>
napisał:

**Comment by psprint** [comment]
How about following plugin: https://github.com/tymm/zsh-directory-history?

**Comment by kadaan** [comment]
I have this almost working correctly with https://github.com/kadaan/per-directory-history.  I can toggle between global and current directory using ^g.  In have a prompt segment in P10K that shows a "world" icon when in global or a "folder" icon when in current directory history. HSMW will correctly show the available history based on the history state (global or current directory) when it is opened.  The only thing missing is the figuring out how to "reset" the open HSMW prompt so that it's list will repopulate to reflect the history state (global or current directory).

@psprint What is the correct way to trigger HSMW to redraw the current view?

**Comment by kadaan** [comment]
@psprint Another thing that would be really helpful is if you know of an easy way to detect if the HSMW prompt is "open".
